gateway/gateway.go

```diff
@@ -27,38 +26,45 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
-type Gateway struct {
-	LocalGatewayID           lorawan.EUI64
-	NetworkGatewayID         lorawan.EUI64
-	PrivateKey               *ecdsa.PrivateKey
-	PublicKey                *ecdsa.PublicKey
-	PublicKeyBytes           []byte
-	CompressedPublicKeyBytes []byte
-	Owner                    common.Address
+// ThingsIxID is the gateways public key without the leading `0x02`.
+type ThingsIxID [32]byte
+
+func (id ThingsIxID) String() string {
+	return fmt.Sprintf("%x", id[:])
 }
 
-// ID is the identifier as which the gateway is registered in the gateway registry.
-func (gw Gateway) ID() [32]byte {
-	var id [32]byte
-	copy(id[:], gw.CompressedPublicKeyBytes)
-	return id
+// Gateway represents a ThingsIX gateway
+type Gateway struct {
+	// LocalID is the gateway ID as used in the communication between gateway
+	// and ThingsIX forwarder and is usually derived from the gateways hardware.
+	LocalID lorawan.EUI64
+	// NetId is the gateway id as used in the communication between the
+	// forwarder and the ThingsIX network.
+	NetID lorawan.EUI64
```

Comment:
Don't use NetID for the NetworkID as it conflicts with the LoRaWAN NetID naming. I would propose using NetworkID

(the NetID is used throughout the code, but it all originates to here).